python/ambassador/envoy/v2/v2listener.py

```diff
@@ -954,46 +954,34 @@ def generate(cls, config: 'V2Config') -> None:
                 # Make sure we have a listener on the right port for this.
                 listener = listeners_by_port[irlistener.insecure_addl_port]
 
-                # ...and make sure we have a VHost for '*'.
-                if '*' not in listener.vhosts:
-                    # This insecure vhost can _only_ have a hostname of "*", since (by definition)
-                    # there is no SNI associated with it.
-                    #
-                    # Also, no, it is not a bug to have action=None. There is no secure action
-                    # for this vhost.
-                    listener.make_vhost(name="redirector",
-                                        hostname="*",
-                                        context=None,
-                                        secure=False,
-                                        action=None,
-                                        insecure_action=irlistener.insecure_action,
-                                        use_proxy_proto=irlistener.use_proxy_proto)
-
-        if config.ir.edge_stack_allowed:
-            # If we're running Edge Stack, make sure we have a listener on port 8080, so that
-            # we have a place to stand for ACME.
-            listener = listeners_by_port[8080]
-
-            # Given the listener, if it has no vhost for '*', add one that rejects everything.
-            # The ACME hole-puncher will override the reject for ACME, and nothing else will
-            # get through.
-            if '*' not in listener.vhosts:
-                # Remember, it is not a bug to have action=None. There is no secure action
+                if irlistener.insecure_addl_port not in first_irlistener_by_port:
+                    first_irlistener_by_port[irlistener.insecure_addl_port] = irlistener
+
+                # This insecure vhost can _only_ have a hostname of "*", since (by definition)
+                # there is no SNI associated with it.
+                #
+                # Also, no, it is not a bug to have action=None. There is no secure action
                 # for this vhost.
-                listener.make_vhost(name="forced-8080",
-                                    hostname="*",
+                listener.make_vhost(name=vhostname,
+                                    hostname=vhostname,
```

Comment:
The comment makes me think that this should be `hostname="*"`.  What's going on?

Reply:
The comment is wrong, but we also need a check to make sure that the `VHost` doesn't already exist. I'll add that shortly.